### PR TITLE
Correcting some documentation

### PR DESCRIPTION
**Pull request by 42jeremy**

### CODE REVIEW

File: memgraph_versioned_docs/version-2.2.1/how-to-guides/streams/kafka/implement-transformation-module.md [hunk 1]
```diff
@@ -192,7 +192,6 @@ So let's create `c_transformation.cpp` and start to populate it!
 ```cpp
 #include <exception>
 #include <string>
-#include <string_view>
```
Comment (antaljanosbenjamin) [inline memgraph_versioned_docs/version-2.2.1/how-to-guides/streams/kafka/implement-transformation-module.md:195]: I think these changes should be primarily done on the current version (in the `docs` folder) and than copy back to every version up-to 2.0.0 where the MGP API was rewamped.

File: cypher-manual/clauses/union.md [hunk 2]
```diff
@@ -59,7 +59,7 @@ Output:
 
 ## 2. Combine queries and remove duplicates
 
-To combine two or more queries and return their results without removing duplicates, use the `UNION` clause without `ALL`.
+To combine two or more queries and return their results while removing duplicates, use the `UNION` clause without `ALL`.
```
Comment (antaljanosbenjamin) [inline cypher-manual/clauses/union.md:62]: The smaller grammar/typo fixes should be also copied back to all of the released version in which they exist. Am I right @g-despot ?

File: memgraph_versioned_docs/version-2.2.1/how-to-guides/streams/kafka/implement-transformation-module.md [hunk 3]
```diff
@@ -201,44 +200,94 @@ const std::string query_part_2{"', payload: '"};
...
+  mgp_value *null_value = NULL;
+  if (mgp_error::MGP_ERROR_NO_ERROR !=
+      mgp_value_make_null(memory, &null_value)) {
+    return;
+  }
 
-  auto *null_value = mgp_value_make_null(memory);
   try {
-    auto messages_size = mgp_messages_size(messages);
+    size_t messages_size;
```
Comment (antaljanosbenjamin) [inline memgraph_versioned_docs/version-2.2.1/how-to-guides/streams/kafka/implement-transformation-module.md:243]:
```suggestion
    size_t messages_size{0};
```

File: memgraph_versioned_docs/version-2.2.1/how-to-guides/streams/kafka/implement-transformation-module.md [hunk 4]
```diff
@@ -201,44 +200,94 @@ const std::string query_part_2{"', payload: '"};
...
+    if (mgp_error::MGP_ERROR_NO_ERROR !=
+        mgp_messages_size(messages, &messages_size)) {
+      return;
+    }
```
Comment (antaljanosbenjamin) [inline memgraph_versioned_docs/version-2.2.1/how-to-guides/streams/kafka/implement-transformation-module.md:247]: If you return here, then `null_value` will be leaked. Maybe get the size of `messages` before creating `null_value`.

File: memgraph_versioned_docs/version-2.2.1/how-to-guides/streams/kafka/implement-transformation-module.md [hunk 5]
```diff
@@ -201,44 +200,94 @@ const std::string query_part_2{"', payload: '"};
...
-                     mgp_message_payload_size(&message)} +
-         query_part_3 + mgp_message_topic_name(&message) + query_part_4;
+std::string create_query(mgp_message &message, struct mgp_result *result) {
+  int64_t timestamp{0};
+  if (mgp_error::MGP_ERROR_NO_ERROR !=
+      mgp_message_timestamp(&message, &timestamp)) {
+    throw "Internal error!";
+  }
+
+  const char *payload;
```
Comment (antaljanosbenjamin) [inline memgraph_versioned_docs/version-2.2.1/how-to-guides/streams/kafka/implement-transformation-module.md:210]:
```suggestion
  const char *payload{nullptr};
```

File: memgraph_versioned_docs/version-2.2.1/how-to-guides/streams/kafka/implement-transformation-module.md [hunk 6]
```diff
@@ -201,44 +200,94 @@ const std::string query_part_2{"', payload: '"};
...
+  if (mgp_error::MGP_ERROR_NO_ERROR !=
+      mgp_message_payload(&message, &payload)) {
+    throw "Internal error!";
+  }
+
+  size_t payload_size;
```
Comment (antaljanosbenjamin) [inline memgraph_versioned_docs/version-2.2.1/how-to-guides/streams/kafka/implement-transformation-module.md:216]:
```suggestion
  size_t payload_size{0};
```

File: memgraph_versioned_docs/version-2.2.1/how-to-guides/streams/kafka/implement-transformation-module.md [hunk 7]
```diff
@@ -201,44 +200,94 @@ const std::string query_part_2{"', payload: '"};
...
+
+      const auto query = create_query(*message, result);
+
+      mgp_result_record *record{nullptr};
+      if (mgp_error::MGP_ERROR_NO_ERROR !=
+          mgp_result_new_record(result, &record)) {
+        break;
+      }
+
+      mgp_value *query_value = NULL;
```
Comment (antaljanosbenjamin) [inline memgraph_versioned_docs/version-2.2.1/how-to-guides/streams/kafka/implement-transformation-module.md:265]:
```suggestion
      mgp_value *query_value{nullptr};
```

**Review by antaljanosbenjamin — CHANGES_REQUESTED** [review]

File: memgraph_versioned_docs/version-2.2.1/how-to-guides/streams/kafka/implement-transformation-module.md (hunk 4 above)
Comment (42jeremy) [reply memgraph_versioned_docs/version-2.2.1/how-to-guides/streams/kafka/implement-transformation-module.md:247]: Good catch, thanks

File: memgraph_versioned_docs/version-2.2.1/how-to-guides/streams/kafka/implement-transformation-module.md (hunk 1 above)
Comment (42jeremy) [reply memgraph_versioned_docs/version-2.2.1/how-to-guides/streams/kafka/implement-transformation-module.md:195]: Will do.

File: cypher-manual/clauses/union.md (hunk 2 above)
Comment (42jeremy) [reply cypher-manual/clauses/union.md:62]: Is this cypher-manual documentation dependent on specific release? A quick search give me a single hit for the phrase "To combine two or more queries and return their results without removing duplicates, use the `UNION` clause without `ALL`.".
(I have looked in /docs)

File: memgraph_versioned_docs/version-2.2.1/how-to-guides/streams/kafka/implement-transformation-module.md (hunk 1 above)
Comment (g-despot) [reply memgraph_versioned_docs/version-2.2.1/how-to-guides/streams/kafka/implement-transformation-module.md:195]: Benjo is correct, you only need to apply the changes in the `/docs` directory, and the DX team will propagate them where required. Usually, we only apply it to the latest version (currently 2.3.0), but if it's only minor changes (copy/paste friendly), then we go even further down the versions.

Comment (42jeremy) [reply memgraph_versioned_docs/version-2.2.1/how-to-guides/streams/kafka/implement-transformation-module.md]: I originally miss-understood, my bad. It should be correct now, only docs/how-to-guides/streams/kafka/implement-transformation-module.md should have the changes.

**Review by g-despot — APPROVED** [review]

**Pull request merged**